packages/clay-drop-down/src/DropDownWithAdvancedItems.tsx

```diff
@@ -0,0 +1,247 @@
+
+	return (
+		<ClayDropDown.Item>
+			<ClayForm.Checkbox
+				{...otherProps}
+				checked={value}
+				onChange={() => {
+					setValue(val => !val);
+					onChange(!value);
+				}}
+			/>
+		</ClayDropDown.Item>
+	);
+};
+
+const Item: React.FunctionComponent<IItem> = (props) => (
+	<ClayDropDown.Item
+		{...props}
+	>
+		{props.label}
+	</ClayDropDown.Item>
+);
+
+const Group: React.FunctionComponent<IAdvancedItem & {spritemap?: string}> = ({ label, items, spritemap }) => (
+	<>
+		<ClayDropDownGroup header={label} />
+		{items && (
+			<DropDownContent
+				items={items}
+				spritemap={spritemap}
+			/>
+		)}
+	</>
+);
+
+interface IRadioContext {
+	checked: string;
+	name?: string;
+	onChange: (value: string) => void;
+}
+
+const RadioGroupContext = React.createContext({} as IRadioContext);
+
+const Radio: React.FunctionComponent<IAdvancedItem> = ({value = '', ...otherProps}) => {
+	const {checked, onChange, name} = useContext(RadioGroupContext);
+
+	return (
+		<ClayDropDown.Item>
+			<ClayForm.Radio
+				{...otherProps}
+				checked={checked === value}
+				inline
+				name={name}
+				onChange={() => onChange(value as string)}
+				value={value as string}
+			/>
+		</ClayDropDown.Item>
+	);
+};
+
+const RadioGroup: React.FunctionComponent<IAdvancedItem & {spritemap?: string}> = ({
+	items,
+	label,
+	name,
+	onChange = () => {},
+	spritemap,
+}) => {
+	const [value, setValue] = useState('');
+
+	const params = {
+		checked: value,
+		name,
+		onChange: (value: string) => {
+			onChange(value);
+			setValue(value);
+		},
+	}
+
+	warning(
+		items && items.filter(item => item.type !== 'radio').length === 0,
+		'ClayDropDownWithAdvancedItems -> Items of type `radiogroup` should be used `radio` if you need to use others, it is recommended to use type `group`.'
+	);
+
+	return (
+		<>
+			{label && (
+				<ClayDropDownGroup header={label} />
+			)}
+			{items && (
+				<li aria-label={label} role="radiogroup">
+					<RadioGroupContext.Provider value={params}>
+						<DropDownContent
+							items={items}
+							spritemap={spritemap}
+						/>
+					</RadioGroupContext.Provider>
+				</li>
+			)}
+		</>
+	)
+};
+
+const TYPE_MAP = {
+	checkbox: Checkbox,
+	group: Group,
+	item: Item,
+	radio: Radio,
+	radiogroup: RadioGroup,
+};
+
+const DropDownContent: React.FunctionComponent<IDropDownContentProps> = ({
+	items,
+	spritemap,
+}) => (
+	<ClayDropDown.ItemList>
+		{items.map(({type, separator, ...item}, key) => {
+			const Item = TYPE_MAP[type || 'item'];
+
+			return (
+				<>
+					<Item {...item} spritemap={spritemap} key={key} />
+					{separator && (
+						<Divider key={`divider${key}`} />
+					)}
+				</>
+			);
+		})}
+	</ClayDropDown.ItemList>
+);
+
+export const ClayDropDownWithAdvancedItems: React.FunctionComponent<IProps> = ({
+	action,
+	caption,
+	helpText,
+	items,
+	onSearchValueChange = () => {},
+	searchable,
+	spritemap,
+	trigger,
+}) => {
+	const [searchValue, setSearchValue] = useState('');
```

Comment:
Rather than using internal state here, I think it might be best to make this a controlled component with both `onSearchValueChange` and `searchValue`.

Reply:
Hmm yeah, I think maybe it's even better this way, devs will be able to filter items when needed.